grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -359,19 +361,12 @@ class StockMovementApiController {
         render([data: pendingRequisitionDetails] as JSON)
     }
 
-    /**
-     * Bind the date field value to the date object.
-     *
-     * @param dateObject
-     * @param jsonObject
-     * @param dateField
-     */
     private Date parseDateRequested(String date) {
-        return date ? Constants.EXPIRATION_DATE_FORMATTER.parse(date) : null
+        return DateUtil.asDate(date, Constants.EXPIRATION_DATE_FORMATTER)
     }
 
     private Date parseDateShipped(String date) {
-        return date ? Constants.DELIVERY_DATE_FORMATTER.parse(date) : null
+        return DateUtil.asDate(date, Constants.DELIVERY_DATE_FORMATTER)
```

Comment:
This is a fix for https://pihemr.atlassian.net/browse/OBS-1831. I wanted something real to test my changes against so figured I'd fix the issue while I was at it.